chapter_01/final_exercise_01.md

```diff
@@ -20,27 +19,142 @@
 
 ### Chapter 1: Introduction to Big Data and Hadoop
 
+## Additional questions:
+
+- [x] *Hdfs Architecture*:
+  - [ ] *Hdfs 1.x*: In hadoop version 1, there was only one Namenode and it was the sigle source of failure of the system.
+  - [ ] *Hdfs 2.x*: Hadoop version 2 introduced which included a standby Namenode which in case of failure of the main name node, it replaces it, thus, removing the single source of failure, however, the name node recovery has to be done manually.
+  - [ ] *Hdfs 3.x*: Introduced support for multiple active namenodes which allow for High availability and now, namenode recovery is done automatically.
+
+- [x] *Hdfs VS S3*: 
+  - [ ] *Hdfs*: A distributed file system that allows for storage of files in a hirarchiel FS, and allows for data querying and processing using tools such as MapReduce and Spark, in addition to the ability to utilize the GPU (as of hadoop 3.x) along with spark, however hdfs is usually self managed or managed using external vendors which come with a not so small cost, while the availability of these systems is lower than the availability of S3 which is managed by amazon and and is cheaper.
+        Hdfs also allows for querying on the data using apache Impala, Hive and Hbase 
+  - [ ] *S3* (Simple Storage Service): can only store and retrieved objects by their key / name, no querying can be done on the s3 and it is not a file system, meaning there are no folders, only file names, S3 allows to tag and store some additional metadata along with the file, has a very simple REST api that can be used and is managed by amazon, with other S3 compatible systems such as Minio that are also Object storage.
+        S3 is relatively cheap, as you only pay for what you use, in addition to it being managed by Amazon which have the resources and the knowledge to provide extremely high availability (11 nines) with good performance and a good cost.
```

Comment:
While this is correct for outside vendors, where you pay for a bucket. Let's look at this from an ONPREM perspective, as this is what we are using here. 
Try to compare the two using the following:  
- Price of storage (hint: replications wise).
- Price of hardware - What hardware does s3 require? what hardware does hdfs require?